subprojects/file-watching/src/main/java/org/gradle/internal/watch/registry/impl/WatchableHierarchies.java

```diff
@@ -93,6 +107,35 @@ public SnapshotHierarchy removeWatchedHierarchiesOverLimit(SnapshotHierarchy roo
         return result;
     }
 
+    private SnapshotHierarchy removeUnwatchableFileSystems(SnapshotHierarchy root, WatchMode watchMode, Invalidator invalidator) {
+        return recentlyUsedHierarchies.stream()
+            .flatMap(watchableFileSystemDetector::detectUnsupportedFileSystemsUnder)
+            .reduce(
+                root,
+                (updatedRoot, fileSystem) -> removeUnwatchableFileSystemIfNecessary(updatedRoot, watchMode, fileSystem, invalidator),
+                nonCombining()
+            );
+    }
+
+    private static SnapshotHierarchy removeUnwatchableFileSystemIfNecessary(SnapshotHierarchy root, WatchMode watchMode, FileSystemInfo fileSystem, Invalidator invalidator) {
+        switch (watchMode) {
+            case ENABLED:
+                LOGGER.debug("Watching is not supported for {}{} file system at '{}'",
+                    fileSystem.isRemote() ? "remote " : "",
+                    fileSystem.getFileSystemType(),
+                    fileSystem.getMountPoint());
+                return root;
+            case DEFAULT:
+                LOGGER.info("Watching is not supported for {}{} file system at '{}', dropping related state from the virtual file system",
```

Comment:
Can we make this log actionable? Like saying that if you don't want to drop the contents, you should enable file system watching?

Reply:
Made a change, WDYT?

Reply:
Much better!